tests/Fixer/ClassNotation/FinalInternalClassFixerTest.php

```diff
@@ -294,37 +320,167 @@ public function provideAnonymousClassesCases(): iterable
 /** @internal */
 $a = new class{};',
         ];
+
+        yield [
+            '<?php $object = new /**/ class(){};',
+        ];
     }
 
     public function testConfigureSameAnnotationInBothLists(): void
     {
         $this->expectException(InvalidFixerConfigurationException::class);
-        $this->expectExceptionMessageMatches(
-            sprintf('#^%s$#', preg_quote('[final_internal_class] Annotation cannot be used in both the include and exclude list, got duplicates: "internal123".', '#'))
-        );
+        $this->expectExceptionMessageMatches(sprintf('#^%s$#', preg_quote('[final_internal_class] Annotation cannot be used in both "include" and "exclude" list, got duplicates: "internal123".', '#')));
 
         $this->fixer->configure([
-            'annotation_include' => ['@internal123', 'a'],
-            'annotation_exclude' => ['@internal123', 'b'],
+            'include' => ['@internal123', 'a'],
+            'exclude' => ['@internal123', 'b'],
         ]);
     }
 
     /**
+     * @group legacy
+     */
+    public function testConfigureBothNewAndOldSet(): void
+    {
+        $this->expectException(InvalidFixerConfigurationException::class);
+        $this->expectExceptionMessageMatches(sprintf('#^%s$#', preg_quote('[final_internal_class] Configuration cannot contain deprecated option "annotation_include" and new option "include".', '#')));
+        $this->expectDeprecation('Option "annotation_include" for rule "final_internal_class" is deprecated and will be removed in version 4.0. Use "internal" to configure PHPDoc annotations tags and attributes.');
+
+        $this->fixer->configure([
+            'annotation_include' => ['@internal', 'a'],
+            'include' => ['@internal', 'b'],
+        ]);
+    }
+
+    /**
+     * @param list<string, list<string>> $config
```

Comment:
We cannot nest `lists`?

Reply:
List is an array with sequential numeric keys starting with 0, so you can't have `list<string, X>`, but `list<list<string>>` should be fine. If keys are string, then `array<string, X>` is what you're looking for.

Reply:
ah check, good to know, thanks so much, updated :+1: